src/mono/wasm/Wasm.Build.Tests/NonWasmTemplateBuildTests.cs

```diff
@@ -108,22 +108,20 @@ private void NonWasmConsoleBuild(string config,
         File.WriteAllText(Path.Combine(_projectDir, "Directory.Build.props"), "<Project />");
         File.WriteAllText(Path.Combine(_projectDir, "Directory.Build.targets"), directoryBuildTargets);
 
-        new DotNetCommand(s_buildEnv, _testOutput, useDefaultArgs: false)
-                .WithWorkingDirectory(_projectDir!)
-                .ExecuteWithCapturedOutput("new console --no-restore")
-                .EnsureSuccessful();
+        using DotNetCommand cmd = new DotNetCommand(s_buildEnv, _testOutput, useDefaultArgs: false);
+        cmd.WithWorkingDirectory(_projectDir!)
+            .ExecuteWithCapturedOutput("new console --no-restore")
+            .EnsureSuccessful();
 
-        new DotNetCommand(s_buildEnv, _testOutput, useDefaultArgs: false)
-                .WithWorkingDirectory(_projectDir!)
-                .ExecuteWithCapturedOutput($"build -restore -c {config} -bl:{Path.Combine(s_buildEnv.LogRootPath, $"{id}.binlog")} {extraBuildArgs} -f {targetFramework}")
-                .EnsureSuccessful();
+        cmd.WithWorkingDirectory(_projectDir!)
```

Comment:
Overriding working dir is void if we reuse the instance (which is by the way interesting idea!). 

Does it have any side effects? The `Dispose` of `ToolCommand` disposes the `CurrentProcess`. Which happens only when the process didn't exit, which can't happen here I guess...

Reply:
True, thanks for pointing out. Follow up PR it is.

One side effect: runs will share the `ITestOutputHelper` instance, so when we do `result.Output` we will see output out of both runs. In this case it does not matter, first we only build, then only run. But in tests where we build twice but with different params and the check the output for `Assert.Contains`, it could be a problem.